daemon/algod/api/server/v2/dryrun_test.go

```diff
@@ -1331,3 +1332,95 @@ func TestDryrunCost(t *testing.T) {
 		})
 	}
 }
+
+func TestDebugTxSubmit(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	a := require.New(t)
+	source := `#pragma version 5
+itxn_begin
+int acfg
+itxn_field TypeEnum
+int 1000000
+itxn_field ConfigAssetTotal
+int 3
+itxn_field ConfigAssetDecimals
+byte "oz"
+itxn_field ConfigAssetUnitName
+byte "Gold"
+itxn_field ConfigAssetName
+byte "https://gold.rush/"
+itxn_field ConfigAssetURL
+byte 0x67f0cd61653bd34316160bc3f5cd3763c85b114d50d38e1f4e72c3b994411e7b
+itxn_field ConfigAssetMetadataHash
+itxn_submit
+int 1`
+
+	ops, err := logic.AssembleString(source)
+	require.NoError(t, err)
+	approval := ops.Program
+
+	ops, err = logic.AssembleString("int 1")
+	clst := ops.Program
+	require.NoError(t, err)
+
+	sender, err := basics.UnmarshalChecksumAddress("47YPQTIGQEO7T4Y4RWDYWEKV6RTR2UNBQXBABEEGM72ESWDQNCQ52OPASU")
+	a.NoError(err)
+	app, err := basics.UnmarshalChecksumAddress("6BPQU5WNZMTO4X72A2THZCGNJNTTE7YL6AWCYSUUTZEIYMJSEPJCQQ6DQI")
+	a.NoError(err)
+
+	// make balance records
+	appIdx := basics.AppIndex(100)
+	dr := DryrunRequest{
+		Txns: []transactions.SignedTxn{{
+			Txn: transactions.Transaction{
+				Type: protocol.ApplicationCallTx,
+				Header: transactions.Header{
+					Sender: sender,
+					Fee:    basics.MicroAlgos{Raw: 100},
+					Note:   []byte{1, 2, 3},
+				},
+				ApplicationCallTxnFields: transactions.ApplicationCallTxnFields{
+					ApplicationID: appIdx,
+				},
+			},
+		}},
+		Apps: []generated.Application{
+			{
+				Id: uint64(appIdx),
+				Params: generated.ApplicationParams{
+					Creator:           sender.String(),
+					ApprovalProgram:   approval,
+					ClearStateProgram: clst,
+				},
+			},
+		},
+		Accounts: []generated.Account{
+			{
+				Address:                     sender.String(),
+				Status:                      "Online",
+				Amount:                      10000000,
+				AmountWithoutPendingRewards: 10000000,
+			},
+			{
+				Address:                     app.String(),
+				Status:                      "Offline",
+				Amount:                      10000000,
+				AmountWithoutPendingRewards: 10000000,
+			},
+			{
+				Address: basics.Address{}.String(),
+				Status:  "Offline",
+			},
+		},
+	}
+
+	dr.ProtocolVersion = string(dryrunProtoVersion)
+
+	var response generated.DryrunResponse
+	doDryrunRequest(&dr, &response)
+	require.NoError(t, err)
+	checkAppCallPass(t, &response)
```

Comment:
It seems like we neglected to add inner transactions to the `DryrunTxnResult` API type. @jannotti was there a reason for this, or did we just forgot about it?

Ideally any created inner transactions would be returned in that response and this test could verify they are there and properly formatted.

Reply:
Definitely not on purpose.

Reply:
I am incapable of expressing how much I hate that dryrun lives in its own universe where we have to do everything all over again.